torch/csrc/distributed/c10d/TraceUtils.h

```diff
@@ -475,7 +477,9 @@ struct NCCLTraceBuffer {
 
     // size information for input/output tensors
     c10::SmallVector<int, 4> input_dims_;
+    std::vector<std::string> input_dtypes_;
     c10::SmallVector<int, 4> output_dims_;
+    std::vector<std::string> output_dtypes_;
```

Comment:
what if we push the raw dtype as an integral type? i assume it is a stable enum in torch somewhere?

Reply:
Yeah that'll work. We'll have new fields as follows:
```
input_dtypes: [6]
output_dtypes: [6]
```
We can create a map in the analyzer to give us more meaningful output.